src/measure_theory/function/ae_eq_fun.lean

```diff
@@ -277,12 +277,62 @@ lift_rel_iff_coe_fn.symm
 instance [partial_order β] : partial_order (α →ₘ[μ] β) :=
 partial_order.lift to_germ to_germ_injective
 
-/- TODO: Prove `L⁰` space is a lattice if β is linear order.
-         What if β is only a lattice? -/
+section lattice
 
--- instance [linear_order β] : semilattice_sup (α →ₘ β) :=
--- { sup := comp₂ (⊔) (_),
---    .. ae_eq_fun.partial_order }
+variables [lattice β] [topological_space β] [topological_lattice β] [borel_space β]
+  [second_countable_topology β]
+
+instance : has_sup (α →ₘ[μ] β) :=
+{ sup := λ f g, ae_eq_fun.comp₂ (⊔) measurable_sup f g }
+
+instance : has_inf (α →ₘ[μ] β) :=
+{ inf := λ f g, ae_eq_fun.comp₂ (⊓) measurable_inf f g }
+
+lemma coe_fn_sup (f g : α →ₘ[μ] β) : ⇑(f ⊔ g) =ᵐ[μ] λ x, f x ⊔ g x := coe_fn_comp₂ _ _ _ _
+lemma coe_fn_inf (f g : α →ₘ[μ] β) : ⇑(f ⊓ g) =ᵐ[μ] λ x, f x ⊓ g x := coe_fn_comp₂ _ _ _ _
+
+private lemma le_sup_left (f g : α →ₘ[μ] β) : f ≤ f ⊔ g :=
```

Comment:
I think that these lemmas should be protected, not private. AFAIR, we try to avoid private lemmas.